x/ibc/04-channel/keeper/handshake.go

```diff
@@ -238,6 +238,11 @@ func (k Keeper) ChanOpenAck(
 		panic("cannot find connection")
 	}
 
+	// verify that chainB's proposed version identifier is supported by chainA
+	if err := connectiontypes.VerifyProposedVersion(channel.Version, counterpartyVersion, nil); err != nil {
```

Comment:
note: no support for disallowing nil feature sets. this is a little more tricky for channel version negotiation. I think it would require creating a channel keeper with the application version set inside a keeper field (perhaps a map). this seems like too much overhead for functionality that hasn't actually been requested yet

Reply:
I completely forgot there are `ChanOpen` callbacks. Applications can just throw an error on any version identifier/feature set they disagree with on `ChanOpenAck`.